AbstractDffractometer, AbstractSampleView and inheriting classes

beteva opened the pull request:
This PR replaces #1386.
- Create AbstractDiffractometer class:
  -   inherits from HardwareObject and uses wait_ready, abort methods and the state handling.
  -  defines some methods, properties and the DiffractometerHead, DiffractometerPhase enums.
  -  defines some scan types
  -  in the init method defines standard access to different equipment (motors and nstate equipment).
- Change DiffarctometerMockup accordingly
- Implement MicroDiffractometer for the Arinax MD2S
- Refactor AbstractSampleView and SampleView to include the methods, previously attached to the diffractometer.

Dependant on [mxcubeweb/pull/2000](https://github.com/mxcube/mxcubeweb/pull/2000)

Commit:
AbstractDiffractometer, AbstractSampleView Hardware Objects.
Test suite and configuration files.
Diffractometer convention.

Commit:
dataa collection and XMLRPC HO changes related to the new Abstract classes.

Commit:
MD2 implementation based on the AbstractDiffractometer.

Commit:
MicrodiffScintillator HO. Implement retry for the ExporterMotor. Cleanup

Commit:
Breaking changes documentation

Commit:
Cleanup

beteva committed:
Update abstract_diffractometer_breaking_changes.md

Commit:
Fight with mdformat.

rhfogh commented on abstract_diffractometer_breaking_changes.md:
```diff
@@ -0,0 +1,63 @@
+- accept_centring
+- reject_centring
+
+#### The centring motors are defined in to sample_view and not diffractometer any more.
```
```suggestion
#### The centring motors are defined in sample_view and not diffractometer any more.
```

rhfogh commented on abstract_diffractometer_breaking_changes.md:
```diff
@@ -0,0 +1,63 @@
+In mxcubecore there are two classes, which could be seen as approximate abstraction for controlling a diffractometer - ***MiniDiff*** and ***GenericDiffractometer***. These classes handle also some sample view and sample centring functionalities. It exists as well an ***AbstractSampleView*** class, which contains only few methods, used to handle the sample visualisation.
```
```suggestion
In mxcubecore there are two classes, which could be seen as approximate abstractions for controlling a diffractometer - ***MiniDiff*** and ***GenericDiffractometer***. These classes handle also some sample view and sample centring functionalities. There is also an ***AbstractSampleView*** class, which contains only few methods, used to handle the sample visualisation.
```

rhfogh commented on diffractometer.md:
```diff
@@ -0,0 +1,39 @@
+Within the concept of the AbstractDiffractometer there are certain number of roles, which are fixed and define a corresponding motor or nstate actuator object.
+
+This allows to use them in a standard way by the MXCuBE application (web or Qt).
```
```suggestion
This allows them to be used in a standard way by the MXCuBE application (web or Qt).
```

rhfogh commented on diffractometer.md:
```diff
@@ -0,0 +1,39 @@
+- **phiy** - alignment table y axis
+- **phiz** - alignment table z axis
+- **sample_horizontal** - x axis, combination of sampx and sampy. Equivalent to sampx at omega=0 and sampy for omega=90.
+- **samle_vertical** - y axis, combination of sampx and sampy. Equivalent to sampy at omega=0 and sampx for omega=90.
```
```suggestion
- **sample_vertical** - y axis, combination of sampx and sampy. Equivalent to sampy at omega=0 and sampx for omega=90.
```

Commit:
Brush up mockup files

beteva committed:
Update abstract_diffractometer_breaking_changes.md

Co-authored-by: Rasmus H. Fogh <rhfogh@globalphasing.com>

beteva committed:
Update abstract_diffractometer_breaking_changes.md

Co-authored-by: Rasmus H. Fogh <rhfogh@globalphasing.com>

beteva committed:
Update diffractometer.md

Co-authored-by: Rasmus H. Fogh <rhfogh@globalphasing.com>

beteva committed:
Update diffractometer.md

Co-authored-by: Rasmus H. Fogh <rhfogh@globalphasing.com>

beteva commented:
Thanks for correcting my spelling @rhfogh

Commit:
Remove obsolete method. Replace methods.

Commit:
Improve _is_beam. get_value_motors instead of get_positions

beteva committed:
Update Beamline.py

Fix typo.

Commit:
Update get_centred_point_from_coord. Add auto_centring_done.

Commit:
Change do_mesh_scan parameters order. Get the dead_time from the detector.

Commit:
Add some missing methods. Improve parameters check.

Commit:
Add accept_centring for automatic centring.
Add test to emit loadedSampleChanged event in _set_loaded_sample.
Remove result from XMLRPC shape_history_get_grid.
Use nb_of_frames in the MD scans if configured.

Commit:
DetectorCover class to handle optional detector cover opening/closing.
Close detector cover when scintillator is IN.
Bug fix in EMBLFFlexHCD

Commit:
Fix from merging.

Commit:
Brush up dead_time

Commit:
Apply SonarQube recommendations.

Commit:
SonarQube and ruff do not agree on this.

marcus-oscarsson committed:
[skip update-version] - Bump version from 1.449.0 to 2.0.0
